sdk/python/kfp/components/structures/kubernetes/v1.py

```diff
@@ -0,0 +1,455 @@
+__all__ = [
```

Comment:
>It seems you are trying to override all objects from kubernetes package 
>Do we have to?

Unfortunately, yes. Those objects lack deserialization support. They also lack the information about which parameters are optional and do not check the passed values for correctness.

>Is it possible to provide a generic converter that takes a k8s object and serialize it using some OSS library?

Serializers are easy. Deserializers are much harder in Python. I've tried several projects and they either do not have any deserialization (e.g. swagger) or do not follow the schema properly (e.g. quicktype.io). Also, the generated code is much bigger than the classes you see here. I found couple of good type validation libraries, but they're Python 3.6+/3.7+ only since typing support is lacking in 3.5.

>I am not sure how we can keep up the kubernetes new versions

I think we can keep up. The core APIs are pretty stable.

Regarding the future, I'm open to switching to a different library, but for now I consider this the most appropriate solution.